rpc/v6/pending_data_wrapper_test.go

```diff
@@ -0,0 +1,99 @@
+package rpcv6_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NethermindEth/juno/clients/feeder"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/mocks"
+	rpc "github.com/NethermindEth/juno/rpc/v6"
+	adaptfeeder "github.com/NethermindEth/juno/starknetdata/feeder"
+	"github.com/NethermindEth/juno/sync"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func TestPendingDataWrapper(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+	mockSyncReader := mocks.NewMockSyncReader(mockCtrl)
+	n := utils.HeapPtr(utils.Sepolia)
+	mockReader := mocks.NewMockReader(mockCtrl)
+	handler := rpc.New(mockReader, mockSyncReader, nil, "", n, nil)
+
+	client := feeder.NewTestClient(t, n)
+	gw := adaptfeeder.New(client)
+
+	latestBlockNumber := uint64(56377)
+	latestBlock, err := gw.BlockByNumber(t.Context(), latestBlockNumber)
+	require.NoError(t, err)
+
+	t.Run("Returns pending block data when starknet version < 0.14.0", func(t *testing.T) {
+		expectedPending := sync.NewPending(latestBlock, nil, nil)
+		mockSyncReader.EXPECT().PendingData().Return(
+			expectedPending.AsPendingData(),
+			nil,
+		)
+		pending, err := handler.PendingData()
+		require.NoError(t, err)
+		require.Equal(t, expectedPending, &sync.Pending{
+			Block:       pending.GetBlock(),
+			StateUpdate: pending.GetStateUpdate(),
+			NewClasses:  pending.GetNewClasses(),
+		})
+	})
+	t.Run("Returns empty pending block when starknet version >= 0.14.0", func(t *testing.T) {
```

Comment:
We should return latest, meaning the last block included in the blockchain (if pre-confirmed will become block `n` we should return `n-1`.